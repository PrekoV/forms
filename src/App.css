.App, .ReduxForm, .FormikForm  {
	height: 100vh;
	width: 100%;
	display: flex;
	justify-content: center;
	align-items: center
}

.App-wrapper, .ReduxForm-wrapper, .FormikForm-wrapper  {
	width: 50%;
	max-width: 800px;
	min-width: 300px;
	background-color: #65096B;
	box-shadow: 0 0 10px #000;
}


.modal {
	width: 90%;
	margin: 10px auto;
}

.intro {
	display: flex;
	justify-content: space-around;
	align-items: center;
	flex-direction: column;
	height: 130px;
}

.circle {
	background-color: #EBD3EC;
	border-radius: 50%;
	height: 70px;
	width: 70px;
	text-align: center;
	display: flex;
	justify-content: center;
	align-items: center;
	color: #65096B;
	font-size: 40px
}

.title {
	font-size: 30px;
	color: #EBD3EC;
	text-transform: uppercase;
}

* {
	font-family: 'Raleway';
	box-sizing: border-box;
	color: #EBD3EC;
}

.input-wrapper {
	border: 2px solid #EBD3EC;
	width: 100%;
	margin: 10px auto;
	width: 100%
}

input, select {
	border: none;
	outline: none;
	background-color: #65096B;
	padding: 10px;
	width: 100%;
	color: #EBD3EC;
	font-size: 20px;
	width: 100%
}

.error{
	height: 20px;
	font-size: 14px;
	
}

.error > div {
	color: #FE4848;
}

button {
	border: none;
	outline: none;
	cursor: pointer;
	text-transform: uppercase;
	text-align: center;
	padding: 10px;
	width: 200px;
	transition: .3s;
	border: 1px solid #65096B;
	background-color:  #EBD3EC;
	color: #65096B
}
button:hover {
	background-color: #65096B;
	color:  #EBD3EC;
	transition: .3s;
	border: 1px solid #EBD3EC;
}

form {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	width: 100%
}

.StepFirst, .StepSecond, .StepThird {
	width: 100%
}

.values{
	font-size: 20px; padding: 5px 0
}

.buttons {
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%
}