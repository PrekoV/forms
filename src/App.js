// import React from "react";
// import { Field, reduxForm } from "redux-form";

// const App = props => {
//     console.log(props);
//     return (
//         <form onSubmit={handleSubmit}>

//         </form>
//     );
// };

// export default reduxForm({
//     form: "simple" // a unique identifier for this form
// })(App);

import React, { Component } from "react";
import "./App.css";
import ReduxForm from "./reduxform";
import FormikForm from "./formik";
import  store from './store'
import { Provider } from 'react-redux'

class App extends Component {
    state = {
        render: 0
    };

    render() {
        return (
            <div className="App">
                {!this.state.render && (
                    <>
                        {" "}
                        <button id="a" onClick={() => this.setState({ render: 1 })}>
                            ReduxForm
                        </button>
                        <button id="b" onClick={() => this.setState({ render: 2 })}>
                            Formik
                        </button>{" "}
                    </>
                )}
                {this.state.render === 1 &&  <Provider  store={store}><ReduxForm /></Provider>}
                {this.state.render === 2 && <FormikForm />}
            </div>
        );
    }
}
export default App;
