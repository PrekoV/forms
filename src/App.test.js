import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import FormikForm from "./formik";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Field from "./reduxform/fieldRedux.js";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Reduxform, { ReduxFormClass } from "./reduxform";
import { Provider } from "react-redux";
import store from "./store";

Enzyme.configure({ adapter: new Adapter() });
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("equals", () => {
    expect(2 + 2).toBe(4);
    expect(2 + 2).toEqual(4);
    expect(2 + 2).toBeGreaterThan(1);
    expect(2 + 2).toBeLessThan(5);
    expect(2 + 2).toBeTruthy();
    expect(2 + 2).not.toBeFalsy();
});

test("Checking the ReduxForm Component", () => {
    expect(FormikForm).toBeDefined();
    const component = renderer.create(<FormikForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.props.className).toEqual("FormikForm");
});

test("app test", () => {
    const component = shallow(<App />);
    expect(component.state("render")).toEqual(0);
    component.find("button#a").simulate("click");
    expect(component.state("render")).toEqual(1);
    expect(component.find("ReduxForm").props().form).toEqual("simple");
});
