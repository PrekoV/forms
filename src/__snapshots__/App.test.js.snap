// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking the ReduxForm Component 1`] = `
<div
  className="FormikForm"
>
  <div
    className="FormikForm-wrapper"
  >
    <div
      className="modal"
    >
      <div
        className="intro"
      >
        <div
          className="circle"
        >
          ✧
        </div>
        <div
          className="title"
        >
          authentication
        </div>
      </div>
      <form
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="error"
        />
        <div
          className="StepFirst"
        >
          <div
            className="fields-wrapper"
          >
            <div
              className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root WithStyles(ForwardRef(TextField))-root-1"
            >
              <label
                className="MuiFormLabel-root Mui-required Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="outlined-required"
              >
                email
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-56"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="outlined-required"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="fields-wrapper"
          >
            <div
              className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root WithStyles(ForwardRef(TextField))-root-1"
            >
              <label
                className="MuiFormLabel-root Mui-required Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="outlined-required"
              >
                password
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-56"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="outlined-required"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="buttons"
        >
          <div />
          <button
            disabled={false}
            type="submit"
          >
            next ↪
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Test redux custom field 1`] = `
<div
  className="fields-wrapper"
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root WithStyles(ForwardRef(TextField))-root-57"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-56"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="outlined-required"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Test redux custom field 2`] = `
<div
  className="fields-wrapper"
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root WithStyles(ForwardRef(TextField))-root-57"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-56"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="outlined-required"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`app test 1`] = `ShallowWrapper {}`;
